skglm/experimental/reweighted.py

```diff
@@ -0,0 +1,44 @@
+import numpy as np
+from numpy.linalg import norm
+from skglm.estimators import Lasso
+
+
+class ReweightedLasso(Lasso):
+    def __init__(self, alpha=1., n_reweight=5, reweight_penalty=None, *args, **kwargs):
+        super().__init__(alpha=alpha, *args, **kwargs)
+        self.n_reweight = n_reweight
+        self.reweight_penalty = reweight_penalty or self._reweight_penalty
+        self.loss_history_ = None
+
+    def fit(self, X, y):
+        self.loss_history_ = []
+        n_samples, n_features = X.shape
+        weights = np.ones(n_features)
+        # XXX: dot product X @ w is slow in high-dimension, to be improved
+        objective = (lambda w: np.sum((y - X @ w) ** 2) / (2 * n_samples)
+                     + self.alpha * np.sqrt(norm(w)))
+
+        for l in range(self.n_reweight):
+            # trick: rescaling the weights (XXX: sparse X would become dense?)
+            scaled_X = X / weights
```

Comment:
use weightedL1 penalty instead :)

Reply:
It’s better to parametrize by the inverse of the weights to avoid dividing by zero. Besides once a weight is zero it’s a lot more effective to run the next iteration only on features with a non zero weight. Then you don’t need the eps in the weight computation

Reply:
we have a WeightedL1 penalty so we don't need to divide X 
feature with infinite weights will be ignored by the working set rule

Reply:
the code above does'nt work if you want to solve MCP with reweighting as its derivative vanishes for values greater than alpha * gamma. That's why we have introduced WeightedL1